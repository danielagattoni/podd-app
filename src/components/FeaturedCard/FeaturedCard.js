import React from 'react';
import './FeaturedCard.scss';

const FeaturedCard = ({ featured }) => {
  const { store, pieOfTheDay } = featured;
  return(
   <div className="FeaturedCard">
     <div className="FeaturedCard-name">
        <div className="FeaturedCard-label">{store.displayName}</div>
        <div className="FeaturedCard-label FeaturedCard-rating">{store.rating}â™¥</div>
     </div>
    <div className="FeaturedCard-address">
      <div className="FeaturedCard-label"><a href={`tel:+${store.mobile}`}>{store.mobile}</a></div>
      <div className="FeaturedCard-label">
        <a
          href={`https://www.google.com.au/maps?q=${store.coords.latitude},${store.coords.longitude}`}
          target="_blank">{store.address}
        </a>
      </div>
    </div>
     <div className="FeaturedCard-frame">
       <h3>Pie of the day:</h3>
       <div className="FeaturedCard-pies">
         <div className="FeaturedCard-pie">
           <div className="FeaturedCard-label">{pieOfTheDay.displayName}</div>
           <div className="FeaturedCard-label">{pieOfTheDay.priceString}</div>
           <div className="FeaturedCard-label">{`Stock:${pieOfTheDay.quantity}`}</div>
         </div>
       </div>
     </div>
   </div>
  );
}

export default FeaturedCard;
