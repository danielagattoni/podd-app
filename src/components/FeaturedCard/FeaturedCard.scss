@import "../../styles/config.mixins.scss";
@import "../../styles/config.variables";

.FeaturedCard {
  position: relative;
  display: flex;
  flex-direction: column;
  width: 100%;
  min-height: 10rem;
  margin: $gutter-v/2 $gutter-h/2;
  padding: $gutter-v $gutter-h;
  color: $color-text-darker;
  text-align: center;
  background-color: $color-card;
  border: 3px solid $color-border;
  border-radius: $border-radius-sm;
  box-shadow: -4px 4px 0 $color-shadow;

  &::before {
    position: absolute;
    content: '';
    top: 2px;
    left: 2px;
    width: 99%;
    border-top: 4px solid $color-lightest;
    border-radius: $border-radius-sm;
  }

  @include break(sm) {
    width: 50%;
  }

  @include break(md) {
    width: 35%;
  }

  &-frame {
    display: flex;
    flex-wrap: wrap;
    align-items: center;
    justify-content: center;
    width: 100%;
    margin-top: $gutter-v/2;
  }

  &-name {
    display: flex;
    align-content: space-evenly;
    align-items: center;
    justify-content: space-between;
    font-size: map-get($font-size-primary, md);
  }

  &-address {
    display: flex;
    flex-direction: column;
  }

  &-pies {
    display: flex;
    flex-direction: column;
  }

  &-pie {
    display: flex;
  }

  &-label {
    margin: $gutter-v/4;
    padding: $gutter-v/4;
    color: $color-text-darker;
    background-color: $color-lightest;
    border: 2px solid lighten($color-cta, 20%);
    border-radius: 4px;
  }

  &-rating {
    color: $color-accent-primary;
  }

}
