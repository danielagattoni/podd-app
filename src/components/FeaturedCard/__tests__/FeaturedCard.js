import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import FeaturedCard from 'components/FeaturedCard/FeaturedCard.js';

const renderer = new ShallowRenderer();

describe('FeaturedCard component', () => {

  it('displays FeaturedCard component', () => {
    const component = renderer.render(
      <FeaturedCard
         key={1}
         featured={{
           store: {
             id: 1,
             displayName: "Eleven-7",
             address: "410 Collins Street",
             mobile: "0481233123",
             coords: {
               latitude: "-37.816981",
               longitude: "144.9584893"
             },
             rating: 10
           },
           pieOfTheDay: {
             displayName: "Beef",
             quantity: 5,
             priceString: "$3.00",
           }
         }} />
       );
    expect(component).toMatchSnapshot();
  });
});
