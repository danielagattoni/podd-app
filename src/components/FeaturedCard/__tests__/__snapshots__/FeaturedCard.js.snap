// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FeaturedCard component displays FeaturedCard component 1`] = `
<div
  className="FeaturedCard"
>
  <div
    className="FeaturedCard-name"
  >
    <div
      className="FeaturedCard-label"
    >
      Eleven-7
    </div>
    <div
      className="FeaturedCard-label FeaturedCard-rating"
    >
      10
      â™¥
    </div>
  </div>
  <div
    className="FeaturedCard-address"
  >
    <div
      className="FeaturedCard-label"
    >
      <a
        href="tel:+0481233123"
      >
        0481233123
      </a>
    </div>
    <div
      className="FeaturedCard-label"
    >
      <a
        href="https://www.google.com.au/maps?q=-37.816981,144.9584893"
        target="_blank"
      >
        410 Collins Street
      </a>
    </div>
  </div>
  <div
    className="FeaturedCard-frame"
  >
    <h3>
      Pie of the day:
    </h3>
    <div
      className="FeaturedCard-pies"
    >
      <div
        className="FeaturedCard-pie"
      >
        <div
          className="FeaturedCard-label"
        >
          Beef
        </div>
        <div
          className="FeaturedCard-label"
        >
          $3.00
        </div>
        <div
          className="FeaturedCard-label"
        >
          Stock:5
        </div>
      </div>
    </div>
  </div>
</div>
`;
